features/localstorage.yml

```diff
@@ -0,0 +1,31 @@
+name: localStorage and sessionStorage
+description: The `localStorage` and `sessionStorage` APIs store data as key-value pairs. While `localStorage` persists across sessions, `sessionStorage` data is discarded when the page is closed.
+spec: https://storage.spec.whatwg.org/
+group: storage
+caniuse: namevalue-storage
+# Computing from here aligns the feature with Caniuse
+status:
+  compute_from: api.Storage
+compat_features:
+  - api.Navigator.storage
+  - api.Storage
+  - api.Storage.clear
+  - api.Storage.getItem
+  - api.Storage.key
+  - api.Storage.length
+  - api.Storage.removeItem
+  - api.Storage.setItem
+  - api.StorageEvent
+  - api.StorageEvent.key
+  - api.StorageEvent.newValue
+  - api.StorageEvent.oldValue
+  - api.StorageEvent.storageArea
+  - api.StorageEvent.url
+  - api.StorageManager
+  - api.StorageManager.estimate
+  - api.StorageManager.persist
+  - api.StorageManager.persisted
```

Comment:
Note that these BCD keys (but also `navigator.storage` and `storage_event`, I think) are not only useful for local/sessionStore APIs, but apply to other forms of storage that an origin can use too.

For example, a developer can use the `navigator.storage.estimate()` promise to discover how much their origin currently uses across IndexedDB, Cache API, and File System API.

I wrote https://patrickbrosset.com/articles/2023-01-17-web-storage/ and https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria, so I have fairly up to date understanding of this area. And I think we should split the StorageManager stuff into its own feature.

In fact, it's not clear to me that the StorageManager actually applies to local/sessionStorage at all. Because these browser-storage mechanisms (as well as cookie) predate the StorageManager API, browsers didn't fold them into the way the StorageManager API handles origin data from the very start. They do intend to manage all types of storage the same way eventually, but I'm not certain this has been done.

Reply:
I had it split out initially as the usage and use-cases are distinct enough from the act of using storage that it seemed like a standalone feature. So whether it does or does not apply to local/sessionStorage, +1 to a separate feature. @ddbeck thoughts?

Reply:
Yeah, sorry for not catching this on initial review myself. StoragManager _does_ seem like a feature, with either `navigator.storage` or `navigator.storage.estimate()` as the entry point for the description. Probably something like, "The `navigator.storage` API provides information about the availability and persistence of data stored by the browser."

(I believe `navigator.storage.getDirectory()` is already part of another feature, for OPFS.)

Reply:
Split out StorageManager again. Is baseline low because of `estimate`, but we could compute from the main feature if that seems like an addition that shouldn't hold the feature back. I erred on the conservative side here because it's a good chunk of the functionality of the overall feature.

Reply:
Yeah, I tend to agree on keeping `estimate` part of the equation here. Even if it actually lies to you (for security reasons), it's one of the key features of the whole API.